Gems/Terrain/Assets/Shaders/Terrain/TerrainMacroTextureClipmapGenerationPass.azsl

```diff
@@ -37,39 +35,42 @@ void SampleTexture(uint2 pixelPosition, uint clipmapLevel, out float3 macroColor
         }
 
         float2 macroUvSize = macroMaterialData.m_boundsMax - macroMaterialData.m_boundsMin;
-        macroUvSize.x = -macroUvSize.x;
-        float2 macroUv = (macroMaterialData.m_boundsMin - worldPosition) / macroUvSize;
+        macroUvSize.x = -1.0 / macroUvSize.x;
+        macroUvSize.y = 1.0 / macroUvSize.y;
+        float2 macroUv = (macroMaterialData.m_boundsMin - worldPosition) * macroUvSize;
```

Comment:
I think these are equivalent, just curious what prompted the change.

Reply:
macroUvSize was used multiple times. So cache it to save rcp calculation according to Jeremy. I forgot to change other places.